src/Microsoft.IdentityModel.Tokens/Json/JsonWebKeySetSerializer.cs

```diff
@@ -136,9 +136,11 @@ public static string Write(JsonWebKeySet jsonWebKeySet)
         /// </summary>
         /// <param name="writer"></param>
         /// <param name="jsonWebKeySet"></param>
-        public static void Write(ref Utf8JsonWriter writer, JsonWebKeySet jsonWebKeySet)
+        /// <param name="shouldWriteObject"></param>
+        public static void Write(ref Utf8JsonWriter writer, JsonWebKeySet jsonWebKeySet, bool shouldWriteObject = true)
```

Comment:
changes to methods with default parameters on public methods leads to a compile time break or runtime error.
Best not to use them.

What is the need for the bool?

Reply:
To opt-out the calling to `writer.WriteStartObject` and `writer.WriteEndObject`. The use cases of `shouldWriterObject: true` as follow:

```c#
jsonWebKeySetSerializer.Write(ref writer, 
                              config.jsonWebKeySet, 
                              shouldWriteObject:true)
                              
JSON generated by writer:
{
   "keys": [
      { /* Json Web Key 1 */ },
      { /* Json Web Key 2 */ },
      /* ... */
   ]
}
```

versus

```c#
jsonWebKeySetSerializer.Write(ref writer, 
                              config.jsonWebKeySet, 
                              shouldWriteObject:false)
                              
JSON generated by writer:
{
   "keys": {
       "keys": [
           { /* Json Web Key 1 */ },
           { /* Json Web Key 2 */ },
           /* ... */
        ]
   }
}
```

Obviously, we want the former use case.

Reply:
I agree and suggested a name change.